### PR TITLE
Powerups

### LINKED ISSUE #42: Doubles bombs propagate through walls
If a wall is in the range of 2 bombs exploding along the same axis. The flames propagate through the wall. 

To replicate place two bombs in a line (BBWW). With a a range of at least 2 both walls will be destroyed.

Looking at the code when a flame-pillar moving in a direction hits a bomb. Propagation of the current flame-pillar should be stopped but trigger the bomb.

### PR DESCRIPTION
**Pull request by charlchi**
Closes #42 #33 #32 #34 issues and includes both PR #47 #44  

Fixes some bomb functionality, adds enemy types and pickups.

(Note: If you can't pickup one of the pickups, that's the door to the next level. Kill all the enemies first. Also the enemies don't kill the player, so you can place bombs directly on enemies which gets funky)

### CODE REVIEW

**Review by TalPat — APPROVED** [review]
Works well

File: models/include/EnemyType.hpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#ifndef _ENEMYTYPE_HPP_
+#define _ENEMYTYPE_HPP_
+
+enum EnemyType{
```
Comment (GJSteyn) [inline models/include/EnemyType.hpp]: Java bracket

File: models/include/IEnemy.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#include <SFML/System.hpp>
+#include <cmath>
+
+enum EnemyMoveState
+{
+	north,
+	east,
+	south,
+	west
+};
```
Comment (GJSteyn) [inline models/include/IEnemy.hpp]: Could use a newline after this bracket

File: models/include/Player.hpp [hunk 3]
```diff
@@ -15,13 +16,11 @@ struct MoveState
...
 class Player
 {
-	sf::Vector2f _position;
-
 	float _playerSpeed;
-
 	bool correctPlayerCellCollision(sf::Vector2i cell);
 
 public:
+	sf::Vector2f _position;
```
Comment (GJSteyn) [inline models/include/Player.hpp]: Don't think this should be public. I think there is a getter function for this.

**Review by GJSteyn — APPROVED** [review]

**Pull request merged**